enhancements/cluster-logging/forward_to_cloudwatch.md

```diff
@@ -0,0 +1,226 @@
+
+Existing fields:
+
+- `url`: Not used in production. Sets the `endpoint` parameter in fluentd for use in testing.
+- `secret`: AWS credentials, keys `aws_access_key_id` and `aws_secret_access_key`.
+
+**Note**: The installer UI (Addon or OLM) can get AWS credentials from a `cloudcredential.openshift.io/v1`.
+The user only has to provide a `region` to enable cloudwatch forwarding for a cluster.
+Details are out of scope for this proposal.
+
+#### Nice To Have: more options for groupBy
+
+_NOT REQUIRED for initial implementation, noted here for possible extensions._
+
+The `groupBy` value is really just then name of a meta-data key in the message.
+There is no implementation cost to allowing arbitrary meta-data to be used as a group name.
+However, the choices should be restricted for safety and simplicity.
+
+A "safe" key must have values that:
+
+1. are valid cloudwatch group name strings.
+2. will not generate an excessive number of groups.
+3. are constant for messages in the same *log stream* (streams belong only one group)
+
+The following keys are safe and would be useful:
+
+- kubernetes.labels.`<key>`: Use pod label value with key `<key>`
+- openshift.labels.`<key>`: Use label added by the openshift log forwarder
+
+Other keys should be considered case-by case, for example:
+
+- `message` is definitely *not* safe, fails all safety requirements.
+- `ip_addr` is safe (node cardinality), but debatable if it would ever be useful.
+- `hostname` is safe (node cardinality), and probably more useful than ip_addr but still debatable.
+- etc.
+
+### User Stories
+
+#### I want to forward logs to Cloudwatch instead of a local store
+
+```
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogForwarder"
+spec:
+  outputs:
+   - name: CloudwatchOut
+     type: cloudwatch
+     cloudwatch:
+       region: myregion
+     secret:
+        name: mysecret
+  pipelines:
+  - inputRefs: [application, infrastructure, audit]
+  - outputRefs: [CloudwatchOut]
+```
+
+#### I want to group application logs by namespace
+
+To group by namespace name:
+
+```
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogForwarder"
+spec:
+  outputs:
+   - name: CloudwatchOut
+     type: cloudwatch
+     cloudwatch:
+       region: myregion
+	   groupBy: namespace_name
+     secret:
+        name: mysecret
+  pipelines:
+  - inputRefs: [application, infrastructure, audit]
+  - outputRefs: [CloudwatchOut]
+```
+
+To group by namespace UUID, replace `namespace_name` with `namespace_uuid`.
+
+### Implementation Details
+
+Using the [fluentd cloudwatch plugin][plugin].
+Most of the CRD fields have an obvious corresponding fluentd parameter.
+
+Set `auto_create_stream true` to create streams and groups on the fly.
+
+`log-stream-name` is set to the `containerID` of the log record.
+
+Log group names are set using `log_group_name` for a static `group` field.
+The `log_group_name_key` can select the log category, namespace or label as group name.
```

Comment:
Lines 86-89 seem to imply we can't use labels, is that true?

Reply:
Correct, thanks. Labels are part of "Nice To Have: more options for groupBy", not required for first impl.